### ISSUE TITLE
[BUG] cattle-cluster-agent crashes after upgrade to 2.9.0

### ISSUE THREAD

**Issue by mkl262** [rancher/rancher#46737]
**Rancher Server Setup**
- Rancher version: 2.9.0
- Installation option (Docker install/Helm Chart):
   - helm chart version: 2.9.0
- Proxy/Cert Details: nginx ingress, NLB load balancer

**Information about the Cluster**
- Kubernetes version: v1.28.11-eks-db838b0
- Cluster Type: Downstream, EKS cluster

**Describe the bug**
After upgrading Rancher server from 2.8.5 to 2.9.0, all of my clusters had their agents updated without an issue, but in one EKS cluster the agent is failing to start up after the update, and the cattle-cluster-agent pods are stuck at a crash loop, with the following error:
```
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x2650e75]

goroutine 2678 [running]:
github.com/rancher/steve/pkg/schema/definitions.listGVKModels({0x9061058, 0xc00aec8878}, 0xc016436000, {0x7fe7da8abc98, 0xc000dfddd0})
	/go/pkg/mod/github.com/rancher/steve@v0.0.0-20240709130809-47871606146c/pkg/schema/definitions/handler.go:226 +0x5b5
github.com/rancher/steve/pkg/schema/definitions.(*SchemaDefinitionHandler).Refresh(0xc001b7e720)
	/go/pkg/mod/github.com/rancher/steve@v0.0.0-20240709130809-47871606146c/pkg/schema/definitions/handler.go:107 +0x17a
github.com/rancher/steve/pkg/debounce.(*DebounceableRefresher).RefreshAfter.func1()
	/go/pkg/mod/github.com/rancher/steve@v0.0.0-20240709130809-47871606146c/pkg/debounce/refresher.go:49 +0x106
created by github.com/rancher/steve/pkg/debounce.(*DebounceableRefresher).RefreshAfter in goroutine 2512
	/go/pkg/mod/github.com/rancher/steve@v0.0.0-20240709130809-47871606146c/pkg/debounce/refresher.go:38 +0x13b
```

**Additional context**
See Attached trace and debug log file for the pod.
[cattle-cluster-agent-5b866f744b-742fb.log](https://github.com/user-attachments/files/16659666/cattle-cluster-agent-5b866f744b-742fb.log)

**Comment by MbolotSuse** [comment]
Hey @mkl262 This issue looks like a duplicate of #46213.  On that basis I'm going to close this issue, please keep an eye on that one for updates. If you think differently please feel free to follow up on this issue and I can give it another look.

**Comment by mkl262** [comment]
Hey @MbolotSuse , thank you very much.

**Comment by mkl262** [comment]
Issue resolved after upgrading to 2.9.1
